common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.maps;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.core.webapi.profiles.MapProfile;
+import com.wynntils.features.user.overlays.map.MapFeature;
+import com.wynntils.features.user.overlays.map.PointerType;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.KeyboardUtils;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.wynn.model.CompassModel;
+import net.minecraft.client.KeyMapping;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.sounds.SimpleSoundInstance;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.sounds.SoundEvents;
+import org.lwjgl.glfw.GLFW;
+
+public class MainMapScreen extends Screen {
+    private static final float SCREEN_SIDE_OFFSET = 10;
+    private static final float BORDER_OFFSET = 7;
+
+    private boolean holdingMapKey = false;
+
+    private float renderWidth;
+    private float renderHeight;
+    private float renderX;
+    private float renderY;
+
+    private float renderedBorderXOffset;
+    private float renderedBorderYOffset;
+
+    private float mapWidth;
+    private float mapHeight;
+    private float centerX;
+    private float centerZ;
+
+    private float mapCenterX;
+    private float mapCenterZ;
+
+    // Zoom is the scaling of the map. The bigger the zoom, the less detailed the map becomes.
+    private static final float MIN_ZOOM = 5.0f;
```

Comment:
This is a bit confusing, having "min" larger than "max". One alternative is to replace the zoomFactor below with it's inverse, i.e. 1/zoomFactor. That way min zoom would be about 0.2 and max zoom about 3. 

I'm not sure this makes more sense, but maybe. Can we "ground" the zoom factor into some understandable "truth"? E.g. if zoom factor 1 means max zoomed in or max zoomed out, or one map pixel per screen pixel, or whatever.

Reply:
Sure, I can try to change it.